Userland/Libraries/LibGfx/GIFLoader.cpp

```
@@ -705,8 +698,8 @@ ErrorOr<ImageFrameDescriptor> GIFImageDecoderPlugin::frame(size_t index)
         }
     }
 
-    if (m_context->error_state == GIFLoadingContext::ErrorState::NoError && !decode_frame(*m_context, index)) {
-        if (m_context->state < GIFLoadingContext::State::FrameComplete || !decode_frame(*m_context, 0)) {
+    if (m_context->error_state == GIFLoadingContext::ErrorState::NoError && decode_frame(*m_context, index).is_error()) {
+        if (m_context->state < GIFLoadingContext::State::FrameComplete || decode_frame(*m_context, 0).is_error()) {
```

Same note here about manual error checking / hiding the underlying error